[iOS] Prepare for Cocoapods 1.3 Release

xta0 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#26751 [iOS] Prepare for Cocoapods 1.3 Release**

### Summary

We're going to use the AWS s3 bucket - `s3://ossci-ios` to store the release binary. To release the cocoapods, we can follow the steps below:

1.  Open a fake PR to trigger the CI job that pulls the code from the 1.3.0 tag branch and does the building and uploading.
2. Verify the binary locally  - Run tests on both arm64 and simulator
3. Publish the cocoapods officially

### Test plan

- podspec lint command succeeds
    - `pod spec lint --verbose --allow-warnings --no-clean --use-libraries --skip-import-validation`

Differential Revision: [D17577131](https://our.internmc.facebook.com/intern/diff/D17577131)

xta0 committed:
[iOS] Prepare for Cocoapods 1.3 Release

xta0 committed:
Update on "[iOS] Prepare for Cocoapods 1.3 Release"


### Summary

For the 1.3 release,  we're going to use the AWS s3 bucket - `s3://ossci-ios` to store the binary. For the release process, I have a proposal below:

1.  Open a fake PR to let the CI job pulls the code from the 1.3.0 tag branch and does the building and uploading.
2. Verify the binary locally  - Run tests on both arm64 and simulator
3. Publish the cocoapods officialy

### Test plan

- podspec lint command succeeds
    - `pod spec lint --verbose --allow-warnings --no-clean --use-libraries --skip-import-validation`

dreiss commented on ios/README.md:
````diff
@@ -1,5 +1,28 @@
+#import <LibTorch/LibTorch.h>
+```
+
+For Swift developers, you need to create an Objective-C class as a bridge to call the C++ APIs. We highly recommend you to follow the [Image Classification](https://github.com/pytorch/examples) demo where you can find out how C++, Objective-C and Swift work together. 
````
Can you update this to the URL where your iOS demo will go?

xta0 replied:
The demo app hasn't been landed, I'll update the link later.

dreiss commented on ios/README.md:
```diff
@@ -1,5 +1,28 @@
+
+### Disable Bitcode 
+
+Since PyTorch is not yet built with bitcode support. You need to disable bitcode for your target by selecting the **Build Settings**, searching for **Enable Bitcode** and set the value to **NO**.
```
"bitcode support, you need"

ezyang commented on ios/README.md:
````diff
@@ -1,5 +1,28 @@
+For Objective-C developers, simply just import the umbrella header
+
+```
+#import <LibTorch/LibTorch.h>
````
How come it's `LibTorch/LibTorch.h`? In C++, it's just `torch/torch.h`, why is it different here?

xta0 replied:
`LibTorch.h` is reserved as an umberlla header that should be imported by Objective-C code. 
Later we'll have Objective-C / Swift wrapper code, which will  rely on this umbrella header to create a swift module.

ezyang replied:
Is there some sort of Objective-C convention here that I don't know about? :)

xta0 replied:
Sorry, didn't see the reply. How to turn on the notification bell on Github? Regarding the umbrella header, I found an article - http://blog.cocoapods.org/Pod-Authors-Guide-to-CocoaPods-Frameworks/. Hopefully, it'll be useful.

ezyang replied:
check your notifications at https://github.com/notifications

by default GitHub will "watch" a bunch of repositories for you which is very bad; unwatch from high traffic ones so you only get pings on explicit mentions and issues you're subscribed to

xta0 committed:
Update on "[iOS] Prepare for Cocoapods 1.3 Release"


### Summary

We're going to use the AWS s3 bucket - `s3://ossci-ios` to store the release binary. To release the cocoapods, we can follow the steps below:

1.  Open a fake PR to trigger the CI job that pulls the code from the 1.3.0 tag branch and does the building and uploading.
2. Verify the binary locally  - Run tests on both arm64 and simulator
3. Publish the cocoapods officially

### Test plan

- podspec lint command succeeds
    - `pod spec lint --verbose --allow-warnings --no-clean --use-libraries --skip-import-validation`
